src/libtorch.cc

```diff
@@ -2119,15 +2357,35 @@ ModelInstanceState::ReadOutputTensors(
 
         // Only need an response tensor for requested outputs.
         if (response != nullptr) {
-          TRITONBACKEND_Output* response_output;
+          if (output_tensor_pair.first != -1) {
+            TRITONBACKEND_Output* response_output;
+            RESPOND_AND_SET_NULL_IF_ERROR(
+                &response, TRITONBACKEND_ResponseOutput(
+                               response, &response_output, name.c_str(),
+                               TRITONSERVER_TYPE_BYTES, batchn_shape.data(),
+                               batchn_shape.size()));
+            string_buffer.emplace_back(new std::string());
+            cuda_copy |= SetStringOutputBuffer(
+                &output_list, &response, response_output, tensor_element_cnt,
+                GetCudaStreamByInstanceKind(), string_buffer.back().get());
+          }
+        }
+        if (output_tensor_pair.second != -1) {
+          TRITONBACKEND_State* response_state;
           RESPOND_AND_SET_NULL_IF_ERROR(
-              &response, TRITONBACKEND_ResponseOutput(
-                             response, &response_output, name.c_str(),
+              &response, TRITONBACKEND_StateNew(
+                             &response_state, request, name.c_str(),
                              TRITONSERVER_TYPE_BYTES, batchn_shape.data(),
                              batchn_shape.size()));
+          RESPOND_AND_SET_NULL_IF_ERROR(
+              &response, TRITONBACKEND_StateNew(
```

Comment:
Is this a mistake with a rebase/merge? Looks like it's repeating the call above

Reply:
Good catch! Fixed.